torch/nn/utils/parametrizations.py

```diff
@@ -163,29 +173,16 @@ def right_inverse(self, Q: torch.Tensor) -> torch.Tensor:
             # to use a particular reflection
             A.diagonal(dim1=-2, dim2=-1)[tau == 0.0] *= -1
             return A.mT if transpose else A
-        else:
-            if n == k:
-                # We check whether Q is orthogonal
-                if not _is_orthogonal(Q):
-                    Q = _make_orthogonal(Q)
-                else:  # Is orthogonal
-                    Q = Q.clone()
-            else:
-                # Complete Q into a full n x n orthogonal matrix
-                N = torch.randn(
-                    *(Q.size()[:-2] + (n, n - k)), dtype=Q.dtype, device=Q.device
-                )
-                Q = torch.cat([Q, N], dim=-1)
-                Q = _make_orthogonal(Q)
-            self.base = Q
-
-            # It is necessary to return the -Id, as we use the diagonal for the
-            # Householder parametrization. Using -Id makes:
-            # householder(torch.zeros(m,n)) == torch.eye(m,n)
-            # Poor man's version of eye_like
-            neg_Id = torch.zeros_like(Q_init)
-            neg_Id.diagonal(dim1=-2, dim2=-1).fill_(-1.0)
-            return neg_Id
+
+        self.base = _ensure_tall_square_orthogonal(Q)
+
+        # It is necessary to return the -Id, as we use the diagonal for the
+        # Householder parametrization. Using -Id makes:
+        # householder(torch.zeros(m,n)) == torch.eye(m,n)
+        # Poor man's version of eye_like
```

Comment:
Related on `eye_like`: 
- https://github.com/pytorch/pytorch/issues/9244
- https://github.com/pytorch/pytorch/issues/68703

Reply:
Thanks @vadimkantorov, I'll look at those issues. Consolidating the APIs inside and outside of the `parameterizations` module would also help make parameterizations more approachable I think.